### PR TITLE
Week 1 blogpost

### PR DESCRIPTION
**Pull request by JoaoDell**
Hello everyone, here is my week 1 blogpost. 
image

### CODE REVIEW

File: docs/source/posts/2023/2023-06-05-week-1-joaodellagli.rst [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+they come with some issues that I will explain further below.
+
+
+My Current Problems
+-------------------
+
+The problems I am having with these FBO implementations is first something related to how a FBO works, and second related to how VTK works. 
+In OpenGL, a custom user's FBO needs some things to be complete (usable):
+
+1. At least one buffer should be attatched. This buffer can be the color, depth or stencil buffer.
```
Comment (tvcastillod) [inline docs/source/posts/2023/2023-06-05-week-1-joaodellagli.rst]: `attatched` => `attached`

File: docs/source/posts/2023/2023-06-05-week-1-joaodellagli.rst [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+2. If no color buffer will be attatched then OpenGL needs to be warned no draw or read operations will be done to that buffer. Otherwise, there should be at least one color attachment.
```
Comment (tvcastillod) [inline docs/source/posts/2023/2023-06-05-week-1-joaodellagli.rst]: `attatched` => `attached`

File: docs/source/posts/2023/2023-06-05-week-1-joaodellagli.rst [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+| color_texture.SetFormat(vtk.VTK_RGBA)
+| color_texture.SetMinificationFilter(0) # setting the minfilter as linear
+| color_texture.SetMagnificationFilter(0) # setting the magfilter as linear
+|
+| color_texture.Allocate2D(width, height, 4, vtk.VTK_UNSIGNED_CHAR) # here is where the code stops
+
+In contrast, for some reason, the methods for 3D textures, `Allocate3D <https://vtk.org/doc/nightly/html/classvtkTextureObject.html#aaeefa46bd3a24bf62126512a276819d0>`_ works just fine. 
+I could use it as a workaround, but I do not wish to, as this just does not make sense. 
+
+My second problem relies VTK. As VTK is a library that encapsulates some OpenGL functions in more palatable forms, it comes with some costs.
```
Comment (tvcastillod) [inline docs/source/posts/2023/2023-06-05-week-1-joaodellagli.rst:49]: **relies on** VTK

File: docs/source/posts/2023/2023-06-05-week-1-joaodellagli.rst [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+As mentioned in the last week's blogpost, the goal for that week was to investigate VTK's Framebuffer Object framework.
+An update on that is that indeed, VTK has one more low-level working `FBO class <vtk.org/doc/nightly/html/classvtkOpenGLFramebufferObject.html>`_ that can be used inside FURY, however, 
+they come with some issues that I will explain further below.
+
+
+My Current Problems
+-------------------
+
+The problems I am having with these FBO implementations is first something related to how a FBO works, and second related to how VTK works.
```
Comment (tvcastillod) [inline docs/source/posts/2023/2023-06-05-week-1-joaodellagli.rst:21]: `is` => `are`

File: docs/source/posts/2023/2023-06-05-week-1-joaodellagli.rst [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+2. If no color buffer will be attatched then OpenGL needs to be warned no draw or read operations will be done to that buffer. Otherwise, there should be at least one color attachment.
+3. All attachments should have their memory allocated.
+4. Each buffer should have the same number of samples. 
+
+My first problem relies on the third requirement. VTK's implementation of FBO requires a `vtkTextureObject <https://vtk.org/doc/nightly/html/classvtkTextureObject.html>`_
+as a texture attachment. I figured how to work with this class, however, I cannot allocate memory for it, as its methods for it, `Allocate2D <https://vtk.org/doc/nightly/html/classvtkTextureObject.html#abc91bbf9a3414bded7a132d366ca4951>`_, `Create2D <https://vtk.org/doc/nightly/html/classvtkTextureObject.html#a7e9dd67f377b7f91abd9df71e75a5f67>`_ and `Create2DFromRaw <https://vtk.org/doc/nightly/html/classvtkTextureObject.html#a0e56fe426cb0e6749cc6f2f8dbf53ed7>`_
+does not seem to work. Every time I try to use them, my program stops with no error message nor nothing. 
+For anyone interested in what is happening exacty, below is how I my tests are implemented:
```
Comment (tvcastillod) [inline docs/source/posts/2023/2023-06-05-week-1-joaodellagli.rst]: `exacty` => `exactly`

File: docs/source/posts/2023/2023-06-05-week-1-joaodellagli.rst [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+   :author: João Victor Dell Agli Floriano
+   :tags: google
+   :category: gsoc
+
+
+This Past Week
+--------------
+
+As mentioned in the last week's blogpost, the goal for that week was to investigate VTK's Framebuffer Object framework.
+An update on that is that indeed, VTK has one more low-level working `FBO class <vtk.org/doc/nightly/html/classvtkOpenGLFramebufferObject.html>`_ that can be used inside FURY, however,
```
Comment (tvcastillod) [inline docs/source/posts/2023/2023-06-05-week-1-joaodellagli.rst:14]: The link does not work for me. Try using this one instead https://vtk.org/doc/nightly/html/classvtkOpenGLFramebufferObject.html

File: docs/source/posts/2023/2023-06-05-week-1-joaodellagli.rst [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+In OpenGL, a custom user's FBO needs some things to be complete (usable):
+
+1. At least one buffer should be attached. This buffer can be the color, depth or stencil buffer.
+2. If no color buffer will be attached then OpenGL needs to be warned no draw or read operations will be done to that buffer. Otherwise, there should be at least one color attachment.
+3. All attachments should have their memory allocated.
+4. Each buffer should have the same number of samples. 
+
+My first problem relies on the third requirement. VTK's implementation of FBO requires a `vtkTextureObject <https://vtk.org/doc/nightly/html/classvtkTextureObject.html>`_
+as a texture attachment. I figured how to work with this class, however, I cannot allocate memory for it, as its methods for it, `Allocate2D <https://vtk.org/doc/nightly/html/classvtkTextureObject.html#abc91bbf9a3414bded7a132d366ca4951>`_, `Create2D <https://vtk.org/doc/nightly/html/classvtkTextureObject.html#a7e9dd67f377b7f91abd9df71e75a5f67>`_ and `Create2DFromRaw <https://vtk.org/doc/nightly/html/classvtkTextureObject.html#a0e56fe426cb0e6749cc6f2f8dbf53ed7>`_
```
Comment (tvcastillod) [inline docs/source/posts/2023/2023-06-05-week-1-joaodellagli.rst:30]: I **figure out** how to

**Review by tvcastillod — COMMENTED** [review]
@JoaoDell Overall it's good, I like that you gave a detailed explanation of your problem. I made some comments for you to review.

File: docs/source/posts/2023/2023-06-05-week-1-joaodellagli.rst (hunk 6 above)
Comment (ganimtron-10) [reply docs/source/posts/2023/2023-06-05-week-1-joaodellagli.rst:14]: Yeah it doesn't work for me as well!

**Review by ganimtron-10 — COMMENTED** [review]
Hey @JoaoDell ,
Nice Detailed Blog.
Just look into @tvcastillod's comment as she has covered all of them!

**Comment by JoaoDell** [comment]
Thanks @ganimtron-10 and @tvcastillod for the reviews, I just adressed the issues you pointed out 👍

**Review by skoudoro — APPROVED** [review]
LGTM, thanks @tvcastillod and @ganimtron-10 for the nice review.

Thanks for this blog post.

merging

**Pull request merged**